src/Compilers/CSharp/Portable/Symbols/Source/ParameterHelpers.cs

```diff
@@ -576,7 +578,19 @@ private static void EnsureNullableAttributeExists(CSharpCompilation compilation,
                         break;
 
                     case SyntaxKind.ReadOnlyKeyword:
-                        diagnostics.Add(ErrorCode.ERR_ReadOnlyNotSuppAsParamModDidYouMeanIn, modifier.GetLocation());
+                        if (seenReadonly)
+                        {
+                            addERR_DupParamMod(diagnostics, modifier);
+                        }
+                        else if (previousModifier?.Kind() != SyntaxKind.RefKeyword)
+                        {
+                            diagnostics.Add(ErrorCode.ERR_RefReadOnlyWrongOrdering, modifier);
+                        }
+                        else if (seenRef)
```

Comment:
Do we need this condition?

Reply:
> Do we need this condition?

It makes an observable difference and, in my opinion, improves quality of diagnostics.

Reply:
It looks like one test would be affected: `params ref readonly`. I think you're correct, this check reduces cascaded diagnostics in such scenario.
Then my problem is with how the variable is named, it would be better as "seenValidRef". That said, previous variables seem to suffer the same problem, so it's fine to leave.